src/tree/AksClusterTreeItem.ts

```diff
@@ -0,0 +1,23 @@
+import { AzExtParentTreeItem, AzureTreeItem } from "vscode-azureextensionui";
+import { Resource } from "azure-arm-storage/lib/models";
+import { CloudExplorerV1 } from "vscode-kubernetes-tools-api";
+
+export default class AkClusterTreeItem extends AzureTreeItem {
```

Comment:
>AkClusterTreeItem [](start = 21, length = 17)

It seems that moving from AKSClusterTreeNode to this new class creates a breaking change for extensions having a dependency on vscode-aks-tools.
For example, in our case we are reading the "session" object from the commandTarget, that is now being replaced by "root".
Our approach is very similar to the one documented here: [https://github.com/Azure/vscode-kubernetes-tools/blob/master/docs/extending/cloudexplorer.md#implementing-your-own-commands-for-cloud-resources](https://github.com/Azure/vscode-kubernetes-tools/blob/master/docs/extending/cloudexplorer.md#implementing-your-own-commands-for-cloud-resources)

This might be fine if no other extensions are taking a dependency on this extension (and we have some time to push a fix for our extension :) ), but I don't know whether it is the case.

Reply:
That's a good point; at the time, I didn't realize other extensions were building atop `vscode-aks-tools`.  (I don't see any other references in GitHub, but Dev Spaces isn't currently OSS and I haven't looked at its codebase in several years, so I would have missed it in any case.)

I could try to make it backward compatible with the existing API.  You could also make future updates to the Dev Spaces extension pinned to the new version of `vscode-aks-tools`, to ensure it always works.  That doesn't help older versions of Dev Spaces extension which end up with the new version of the AKS extension, obviously.

Reply:
I've made updates that map (most of) the properties defined in the old interfaces to their respective newer properties.  This appears to resolve the issue with the existing Dev Spaces extension.  The properties that are not mapped did not appear to be exposed directly by the subscription/resource nodes, but also appear to be the more esoteric ones related to the subscription itself that I hope are low risk for breaking back-compat.

Let me know if this seems more reasonable.

Reply:
That looks good to me indeed. I also validated that, with the new version of your changes, our extension works as expected. Thanks for fixing this quickly!